include/seqan3/argument_parser/validators.hpp

```diff
@@ -235,69 +236,87 @@ value_list_validator(std::vector<const char *> const & v) -> value_list_validato
+    path_validator_base & operator=(path_validator_base const &) = default; //!< Defaulted.
+    path_validator_base & operator=(path_validator_base &&) = default;      //!< Defaulted.
+    ~path_validator_base() = default;                                       //!< Defaulted.
+
+    /*!\brief Constructs from a set of valid extensions.
+     * \param[in] extensions The valid extensions to validate for.
+     */
+    path_validator_base(std::vector<std::string> extensions) : extensions{std::move(extensions)}
+    {}
+    //!\}
 
-    /*!\brief Tests whether the filepath \p path ends with a valid extension.
-     * \param path The input value to check.
+public:
+
+    //!\brief Type of values that are tested by validator.
+    using value_type = std::string;
+
+    /*!\brief Tests if the given path is a valid input, respectively output, file or directory.
+     * \param path The path to validate.
      * \throws parser_invalid_argument
      */
     void operator()(std::filesystem::path const & path) const
     {
-        std::string ext{path.extension().string()};
-        ext = ext.substr(std::min(1, std::max(0, static_cast<int>(ext.size()) - 1))); // drop '.' if ext is non-empty
-
-        // extensions were transformed to lower case during construction,so do the same for input path
-        if (!case_sensitive)
-            ext = std::string{ext | view::to_lower};
-
-        if (!(std::find(extensions.begin(), extensions.end(), ext) != extensions.end()))
-            throw parser_invalid_argument(detail::to_string("Extension ", ext, " is not one of ",
-                                                            std::view::all(extensions), "."));
+        // Call derived type validation procedure.
+        static_cast<derived_t const &>(*this).validate_path_impl(path);
+
+        // If no valid extensions are given we can safely return here.
+        if (extensions.empty())
+            return;
+
+        // Check if extension is available.
+        if (!path.has_extension())
+            throw parser_invalid_argument(detail::to_string("The given filename ", path.string(),
+                                                            " has no extension."));
+
+        // Drop the dot.
+        std::string tmp_str = path.extension().string();
+        auto drop_less_ext = tmp_str | view::drop(1);
+
+        // Compares the extensions in lower case.
+        auto cmp_lambda = [&] (std::string const & cmp)
+        {
+            return std::ranges::equal(cmp | view::to_lower, drop_less_ext | view::to_lower);
+        };
+
+        // Check if requested extension is present.
+        if (auto res = std::find_if(extensions.begin(), extensions.end(), cmp_lambda); res == extensions.end())
+        {
+            throw parser_invalid_argument(detail::to_string("Expected on of the following valid extensions: ",
+                                                             std::view::all(extensions), ". Got ",
```

Comment:
```suggestion
                                                             extensions, ". Got ",
```
This should be handled by to-string automatically